presto-tpcds/src/test/java/com/facebook/presto/sql/planner/optimizations/TestTpcdsCostBasedPlan.java

```diff
@@ -0,0 +1,107 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.sql.analyzer.FeaturesConfig.JoinDistributionType;
+import com.facebook.presto.sql.analyzer.FeaturesConfig.JoinReorderingStrategy;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tpcds.TpcdsConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static com.facebook.presto.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+
+/**
+ * This class tests cost-based optimization rules related to joins. It contains unmodified TPCDS queries.
+ * This class is using TPCDS connector configured in way to mock Hive connector with unpartitioned TPCDS tables.
+ */
+public class TestTpcdsCostBasedPlan
+        extends BaseCostBasedPlanTest
+{
+    /*
+     * CAUTION: The expected plans here are not necessarily optimal yet. Their role is to prevent
+     * inadvertent regressions. A conscious improvement to the planner may require changing some
+     * of the expected plans, but any such change should be verified on an actual cluster with
+     * large amount of data.
+     */
+
+    public TestTpcdsCostBasedPlan()
+    {
+        super(() -> {
+            String catalog = "local";
+            Session.SessionBuilder sessionBuilder = testSessionBuilder()
+                    .setCatalog(catalog)
+                    .setSchema("sf3000.0")
+                    .setSystemProperty("task_concurrency", "1") // these tests don't handle exchanges from local parallel
+                    .setSystemProperty(JOIN_REORDERING_STRATEGY, JoinReorderingStrategy.AUTOMATIC.name())
+                    .setSystemProperty(JOIN_DISTRIBUTION_TYPE, JoinDistributionType.AUTOMATIC.name());
+
+            LocalQueryRunner queryRunner = LocalQueryRunner.queryRunnerWithFakeNodeCountForStats(sessionBuilder.build(), 8);
+            queryRunner.createCatalog(
+                    catalog,
+                    new TpcdsConnectorFactory(1),
+                    ImmutableMap.of());
+            return queryRunner;
+        });
+    }
+
+    @Override
+    protected Stream<Query> getQueries()
+    {
+        return IntStream.range(1, 100)
+                .boxed()
+                .flatMap(i -> {
+                    String queryId = format("q%02d", i);
+                    if (i == 14 || i == 23 || i == 24 || i == 39) {
```

Comment:
This looks like something very ugly

Reply:
It looks, but this is how tpcds standard is defined. What do you propose to handle queries `14_1, 14_2, 23_1, 23_2...`?